### DISCUSSION TITLE
Is it possible to hard force stress when training xtts2 model

### DISCUSSION THREAD

**Discussion by brambox** [idiap/coqui-ai-TTS#28] [Q&A]
It seems that no mater how much you train the model it will very often put stress on different positions.  Is there any way using vocab or something to hard force stress position to always be accurate  and don't pull the rng and put the stress what ever it decides it?

**Comment by eginhard** [comment]
The old-school way would be to train a model on phonemes with stress labels, but of course XTTS is not even trained on phonemes. Since the model doesn't directly receive that information, there is no way to force a specific stress and you can only hope it guesses correctly from the context when it's trained on more data.

At inference time, you could try something like capitalising the syllable that should be stressed. During training, you could try to steer the model in that direction by capitalising the stressed syllables there as well.

**Comment by brambox** [comment]
thank you! Isn't cleaners auto converting all to lowercase on inference ?

**Reply by eginhard** [reply]
Right, then this won't work out of the box either.
